### PR TITLE
Add reject message

### PR DESCRIPTION
**Pull request by tamasblummer**
Quite a dumb message,  will probably disappear. Some nodes are still sending it though and not having it creates errors in parser.

### CODE REVIEW

File: src/network/message_network.rs [hunk 1]
```diff
@@ -78,6 +84,66 @@ impl_consensus_encoding!(VersionMessage, version, services, timestamp,
                          receiver, sender, nonce,
                          user_agent, start_height, relay);
 
+#[repr(u8)]
```
Comment (elichai) [inline src/network/message_network.rs]: Why the repr?

Comment (tamasblummer) [reply src/network/message_network.rs]: to be able to convert the enum into u8 with as u8

Comment (elichai) [reply src/network/message_network.rs]: I'll try with 1.22 but with newer rust versions it works just fine:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=88287b6224e3e4b5bb215ef26fe323e7

Comment (elichai) [reply src/network/message_network.rs]: Just tried with 1.22.0 and works.
the repr is about memory representation, not about casting.
repr's are only if you pass C ABI or you transmute to a different one.

Comment (tamasblummer) [reply src/network/message_network.rs]: sounds like a serious problem

Comment (elichai) [reply src/network/message_network.rs]: Why a problem?
I just don't think it's necessary

Comment (stevenroose) [reply src/network/message_network.rs]: I think what he's saying is that `as u8` also works without the `repr`. `repr` is used when you are using the type in a C API. So I agree we don't need it here.

File: src/network/message_network.rs [hunk 2]
```diff
@@ -78,6 +84,66 @@ impl_consensus_encoding!(VersionMessage, version, services, timestamp,
...
+    MALFORMED = 0x01,
+    /// invalid message
+    INVALID = 0x10,
+    /// obsolete message
+    OBSOLETE = 0x11,
+    /// duplicate message
+    DUPLICATE = 0x12,
+    /// nonstandard transaction
+    NONSTANDARD = 0x40,
+    /// an output is belw dust limit
```
Comment (odarboe) [inline src/network/message_network.rs]: typo *below dust limit.

**Comment by tamasblummer** [comment]
this is a trivial already approved but then stopped for an extra #[repr(u8)], now pending since 18 days ... would you please approve @apoelstra @stevenroose

**Comment by tamasblummer** [comment]
I would love to have this into a new release of rust-bitcoin, so I can finally publish dependent projects, such as rust-wallet and get rid of patches in others. Please approve @stevenroose @TheBlueMatt @elichai

**Comment by tamasblummer** [comment]
@dongcarl would you please look at this trivial addition. I really need it to get rid of downstream patches.

**Comment by elichai** [comment]
Sorry I really don't know enough about the p2p to review this easily.
But rustwise it looks pretty good.

**Comment by tamasblummer** [comment]
@stevenroose please, this is trivial and pending since a month on nits

**Comment by tamasblummer** [comment]
@dongcarl thanks a lot
@apoelstra would you please cut a release?

**Pull request merged**